File: outliner/tests/fixtures/java/sample.java
package com.example;

import java.io.IOException;
import java.util.Optional;

/**
 * Represents an animal in the system.
 */
@SuppressWarnings("unused")
public class Animal {

    private final String name;
    private int age;

    /**
     * Creates a new Animal with the given name and age.
     *
     * @param name the animal's name
     * @param age  the animal's age
     */
    public Animal(String name, int age) {
        this.name = name;
        this.age = age;
    }

    /** Returns the animal's name. */
    public String getName() throws IOException {
        return name;
    }

    public int getAge() {
        return age;
    }

    /**
     * Checks whether this animal is older than another.
     */
    public boolean isOlderThan(Animal other) {
        return this.age > other.age;
    }

    @Override
    public String toString() {
        return "Animal(" + name + ", " + age + ")";
    }

    @Deprecated
    public static Animal fromLegacy(String spec) {
        String[] parts = spec.split(":");
        return new Animal(parts[0], Integer.parseInt(parts[1]));
    }

    /**
     * Produces an animal from a list of strings.
     * Multi-line method to test signature merging.
     */
    public static Optional<Animal> fromParts(
            String name,
            int age,
            boolean validate) {
        if (validate && name.isEmpty()) return Optional.empty();
        return Optional.of(new Animal(name, age));
    }

    /** Returns animals grouped by tag; exercises nested-generic signatures. */
    public static Map<String, List<Animal>> groupByTag(
            Map<String, List<Animal>> source,
            boolean deepCopy) {
        return deepCopy ? new java.util.HashMap<>(source) : source;
    }

    // -----------------------------------------------------------------------
    // Nested class
    // -----------------------------------------------------------------------

    /** Metadata associated with an animal record. */
    public static class Metadata {

        private final String source;

        public Metadata(String source) {
            this.source = source;
        }

        public String getSource() {
            synchronized (this) {
                return source;
            }
        }
    }
}

// ---------------------------------------------------------------------------

/** Defines behaviour for a speakable entity. */
public interface Speakable {

    String speak();

    default String shout() {
        return speak().toUpperCase();
    }
}

// ---------------------------------------------------------------------------

/** Possible states of an animal record. */
public enum Status {
    ACTIVE,
    INACTIVE,
    ARCHIVED;

    public boolean isVisible() {
        return this == ACTIVE;
    }
}

// ---------------------------------------------------------------------------

/** Annotation type for tagging animal records. */
public @interface AnimalTag {

    String value() default "";

    int priority() default 0;
}

// ---------------------------------------------------------------------------

/** A 2D point with Euclidean distance. */
public record Point(double x, double y) {

    public double distance() {
        return Math.sqrt(x * x + y * y);
    }
}

File: outliner/tests/fixtures/rs/sample.rs
//! Animal module: types and traits for managing animals.

use std::fmt;
use std::collections::HashMap;

/// An animal with a name and age.
///
/// Created via `Animal::new`.
#[derive(Debug, Clone, PartialEq)]
pub struct Animal {
    /// The animal's name.
    pub name: String,
    age: u32,
}

/// Possible statuses of an animal record.
#[derive(Debug, Clone, PartialEq)]
pub enum Status {
    Active,
    Inactive,
    Archived,
}

/// Trait for entities that can speak.
pub trait Speakable {
    /// Returns the entity's speech.
    fn speak(&self) -> String;

    /// Returns uppercased speech.
    fn shout(&self) -> String {
        self.speak().to_uppercase()
    }
}

impl Animal {
    /// Creates a new Animal.
    pub fn new(name: &str, age: u32) -> Self {
        Animal { name: name.to_string(), age }
    }

    /// Returns the animal's name.
    pub fn name(&self) -> &str {
        &self.name
    }

    /// Constructs an animal from separate parts.
    /// Multi-line signature to test merging.
    pub fn from_parts(
        name: String,
        age: u32,
        validate: bool,
    ) -> Option<Animal> {
        if validate && name.is_empty() {
            return None;
        }
        Some(Animal { name, age })
    }

    /// Returns whether this animal is older.
    pub fn is_older_than(&self, other: &Animal) -> bool {
        self.age > other.age
    }
}

impl Speakable for Animal {
    fn speak(&self) -> String {
        format!("I am {}", self.name)
    }
}

impl fmt::Display for Animal {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        write!(f, "Animal({}, {})", self.name, self.age)
    }
}

/// Type alias for an animal registry.
pub type Registry = HashMap<String, Animal>;

/// Computes factorial of n.
pub fn factorial(n: u64) -> u64 {
    if n == 0 { 1 } else { n * factorial(n - 1) }
}

/// Parses an animal from input using a factory closure.
/// Demonstrates lifetime and generic parameters.
pub fn parse_animal<'a, F>(
    input: &'a str,
    factory: F,
) -> Option<Animal>
where
    F: Fn(&'a str) -> Option<Animal>,
{
    factory(input)
}

/// A generic pair of values.
pub struct Pair<T, U> {
    pub first: T,
    pub second: U,
}

impl<T: Clone, U: Clone> Pair<T, U> {
    /// Creates a new Pair.
    pub fn new(first: T, second: U) -> Self {
        Pair { first, second }
    }

    /// Swaps the pair elements.
    pub fn swap(&self) -> Pair<U, T> {
        Pair { first: self.second.clone(), second: self.first.clone() }
    }
}

/// Internal utilities module.
mod utils {
    /// Doubles the input value.
    pub fn double(x: u32) -> u32 {
        x * 2
    }
}
